stream/stream.go

```diff
@@ -239,57 +240,59 @@ func (s *Client) PublishBuilderBid(ctx context.Context, bid structs.BuilderBidEx
 	timer1.ObserveDuration()
 
 	l := s.Logger.With(log.F{
-		"method":    "publishBuilderBid",
-		"itemType":  "builderBid",
-		"size":      len(b),
+		"method":   "publishBuilderBid",
+		"itemType": "builderBid",
+		// "size":      len(b),
 		"blockHash": bid.BuilderBid().Header().GetBlockHash(),
 	})
 
 	timer2 := prometheus.NewTimer(s.m.Timing.WithLabelValues("publishBuilderBid", "publish"))
 	l.WithField("timestamp", time.Now().String()).Debug("publishing")
-	if err := s.Pubsub.Publish(ctx, s.Config.PubsubTopic+BidTopic, b); err != nil {
+	if err := s.Bids.Publish(ctx, b); err != nil {
 		return fmt.Errorf("fail to encode encode and stream block: %w", err)
 	}
 	l.WithField("timestamp", time.Now().String()).Debug("published")
 	timer2.ObserveDuration()
 
-	s.m.PublishSize.WithLabelValues("publishBuilderBid").Observe(float64(len(b)))
-	s.m.PublishCounter.WithLabelValues("publishBuilderBid").Add(float64(len(b)))
+	// s.m.PublishSize.WithLabelValues("publishBuilderBid").Observe(float64(len(b)))
+	// s.m.PublishCounter.WithLabelValues("publishBuilderBid").Add(float64(len(b)))
 
 	timer0.ObserveDuration()
 	return nil
 }
 
 func (s *Client) PublishBlockCache(ctx context.Context, block structs.BlockAndTraceExtended) error {
+	s.init()
+
 	timer0 := prometheus.NewTimer(s.m.Timing.WithLabelValues("publishCacheBlock", "all"))
 
 	timer1 := prometheus.NewTimer(s.m.Timing.WithLabelValues("publishCacheBlock", "encode"))
 	forkEncoding := toBlockCacheFormat(s.st.ForkVersion(structs.Slot(block.Slot())))
-	b, err := s.encode(block, forkEncoding)
+	msg, err := s.encode(block, forkEncoding)
```

Comment:
Shouldn't this be a a transport function? If you distinct between `Cache` and `Bids` wouldn't that be optimal to pass entire object to publish and let transport internally decide how to encode/carry that?

Reply:
It could, and likely eventually will.  I'd like to do this in incremental steps, though. For now, it seemed sensible to move the payload envelope (`transport.Message`, formerlly `JsonItem`) to the `transport` package and leave the payload encoding where it is.